pkg/v1/tarball/image.go

```diff
@@ -230,7 +232,28 @@ func extractFileFromTar(opener Opener, filePath string) (io.ReadCloser, error) {
 		}
 	}()
 
-	tf := tar.NewReader(f)
+	var tf *tar.Reader
+	magicHeader := make([]byte, len(ggzip.MagicHeader))
+	n, err := f.Read(magicHeader)
+	if n == 0 && err == io.EOF {
+		return nil, errors.New("invalid tar header")
```

Comment:
maybe the error should be reported as "invalid tar or tar.zip header" to match the context, as both formats are supported

Reply:
> maybe the error should be reported as "invalid tar or tar.zip header" to match the context, as both formats are supported

done